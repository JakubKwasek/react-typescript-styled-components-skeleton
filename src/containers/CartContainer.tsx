import * as React from "react";
import { Box, Button } from "grommet";
import { booksCart } from "../data/booksCart";
import { BoxFactory } from "../components/BoxFactory";

enum QuantityOperation {
	"add",
	"remove"
}

export const RemoveFromCart = (id: number): void => {
	console.log(id);
};

export const ChangeQuantity = (id: number, operation: QuantityOperation): void => {
	console.log(id);
};

export const CartContainer = (): JSX.Element => (
	<>
		{
			booksCart.map((book) => (
				<Box
					align="start"
					direction={"row"}
					border={{side: "bottom",}}
					pad="small"
					key={book.id}>
					<BoxFactory
						width={{ min: "10%", }}>
						<img src={book.image} />
					</BoxFactory>
					<BoxFactory
						width={{ min: "30%", }}>
						<>
							<small>Title:</small>
							<p>{book.title}</p>
						</>
					</BoxFactory>
					<BoxFactory
						width={{ min: "20%", }}>
						<>
							<small>Price:</small>
							<p>{book.price}</p>
						</>
					</BoxFactory>
					<BoxFactory
						width={{ min: "20%", }}>
						<>
							<small>Quantity:</small>
							<p>
								<Button
									margin={{ horizontal: "10px",}}
									label="-"
									reverse
									onClick={(): void =>
										ChangeQuantity(book.id, QuantityOperation.remove)}/>
								{book.quantity}
								<Button
									margin={{ horizontal: "10px",}}
									label="+"
									reverse
									onClick={(): void =>
										ChangeQuantity(book.id, QuantityOperation.add)}/>
							</p>
						</>
					</BoxFactory>
					<BoxFactory
						width={{ min: "20%", }}>
						<>
							<small>Action:</small>
							<p>
								<Button
									label="Remove"
									primary
									onClick={(e): void =>
										RemoveFromCart(book.id)} />
							</p>
						</>
					</BoxFactory>
				</Box>
			))
		}
	</>
);
