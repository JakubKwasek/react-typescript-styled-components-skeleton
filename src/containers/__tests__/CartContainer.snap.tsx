import * as React from "react";
import * as renderer from "react-test-renderer";
import { CartContainer } from "../CartContainer";

describe("CartContainer", () => {
	it("renders correctly", () => {
		const tree = renderer
			.create(<CartContainer />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
