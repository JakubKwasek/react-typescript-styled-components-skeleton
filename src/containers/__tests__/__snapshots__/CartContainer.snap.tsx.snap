// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartContainer renders correctly 1`] = `
Array [
  <div
    className="StyledBox-sc-13pk1d4-0 bZxUit"
  >
    <div
      className="StyledBox-sc-13pk1d4-0 hPokmc"
    >
      <img
        src="http://dummyimage.com/50x70.png/ff4444/ffffff"
      />
    </div>
    <div
      className="StyledBox-sc-13pk1d4-0 WHhky"
    >
      <small>
        Title:
      </small>
      <p>
        One False Move
      </p>
    </div>
    <div
      className="StyledBox-sc-13pk1d4-0 bbVmir"
    >
      <small>
        Price:
      </small>
      <p>
        $7.67
      </p>
    </div>
    <div
      className="StyledBox-sc-13pk1d4-0 bbVmir"
    >
      <small>
        Quantity:
      </small>
      <p>
        <button
          className="StyledButton-sc-323bzc-0 fYqBJk"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          type="button"
        >
          -
        </button>
        2
        <button
          className="StyledButton-sc-323bzc-0 fYqBJk"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          type="button"
        >
          +
        </button>
      </p>
    </div>
    <div
      className="StyledBox-sc-13pk1d4-0 bbVmir"
    >
      <small>
        Action:
      </small>
      <p>
        <button
          className="StyledButton-sc-323bzc-0 fErzFK"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          type="button"
        >
          Remove
        </button>
      </p>
    </div>
  </div>,
  <div
    className="StyledBox-sc-13pk1d4-0 bZxUit"
  >
    <div
      className="StyledBox-sc-13pk1d4-0 hPokmc"
    >
      <img
        src="http://dummyimage.com/50x70.png/ff4444/ffffff"
      />
    </div>
    <div
      className="StyledBox-sc-13pk1d4-0 WHhky"
    >
      <small>
        Title:
      </small>
      <p>
        Creation of the Humanoids, The
      </p>
    </div>
    <div
      className="StyledBox-sc-13pk1d4-0 bbVmir"
    >
      <small>
        Price:
      </small>
      <p>
        $9.48
      </p>
    </div>
    <div
      className="StyledBox-sc-13pk1d4-0 bbVmir"
    >
      <small>
        Quantity:
      </small>
      <p>
        <button
          className="StyledButton-sc-323bzc-0 fYqBJk"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          type="button"
        >
          -
        </button>
        1
        <button
          className="StyledButton-sc-323bzc-0 fYqBJk"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          type="button"
        >
          +
        </button>
      </p>
    </div>
    <div
      className="StyledBox-sc-13pk1d4-0 bbVmir"
    >
      <small>
        Action:
      </small>
      <p>
        <button
          className="StyledButton-sc-323bzc-0 fErzFK"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          type="button"
        >
          Remove
        </button>
      </p>
    </div>
  </div>,
  <div
    className="StyledBox-sc-13pk1d4-0 bZxUit"
  >
    <div
      className="StyledBox-sc-13pk1d4-0 hPokmc"
    >
      <img
        src="http://dummyimage.com/50x70.png/5fa2dd/ffffff"
      />
    </div>
    <div
      className="StyledBox-sc-13pk1d4-0 WHhky"
    >
      <small>
        Title:
      </small>
      <p>
        Judith of Bethulia
      </p>
    </div>
    <div
      className="StyledBox-sc-13pk1d4-0 bbVmir"
    >
      <small>
        Price:
      </small>
      <p>
        $6.85
      </p>
    </div>
    <div
      className="StyledBox-sc-13pk1d4-0 bbVmir"
    >
      <small>
        Quantity:
      </small>
      <p>
        <button
          className="StyledButton-sc-323bzc-0 fYqBJk"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          type="button"
        >
          -
        </button>
        1
        <button
          className="StyledButton-sc-323bzc-0 fYqBJk"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          type="button"
        >
          +
        </button>
      </p>
    </div>
    <div
      className="StyledBox-sc-13pk1d4-0 bbVmir"
    >
      <small>
        Action:
      </small>
      <p>
        <button
          className="StyledButton-sc-323bzc-0 fErzFK"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          type="button"
        >
          Remove
        </button>
      </p>
    </div>
  </div>,
  <div
    className="StyledBox-sc-13pk1d4-0 bZxUit"
  >
    <div
      className="StyledBox-sc-13pk1d4-0 hPokmc"
    >
      <img
        src="http://dummyimage.com/50x70.png/dddddd/000000"
      />
    </div>
    <div
      className="StyledBox-sc-13pk1d4-0 WHhky"
    >
      <small>
        Title:
      </small>
      <p>
        23 (23 - Nichts ist so wie es scheint)
      </p>
    </div>
    <div
      className="StyledBox-sc-13pk1d4-0 bbVmir"
    >
      <small>
        Price:
      </small>
      <p>
        $1.16
      </p>
    </div>
    <div
      className="StyledBox-sc-13pk1d4-0 bbVmir"
    >
      <small>
        Quantity:
      </small>
      <p>
        <button
          className="StyledButton-sc-323bzc-0 fYqBJk"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          type="button"
        >
          -
        </button>
        2
        <button
          className="StyledButton-sc-323bzc-0 fYqBJk"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          type="button"
        >
          +
        </button>
      </p>
    </div>
    <div
      className="StyledBox-sc-13pk1d4-0 bbVmir"
    >
      <small>
        Action:
      </small>
      <p>
        <button
          className="StyledButton-sc-323bzc-0 fErzFK"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          type="button"
        >
          Remove
        </button>
      </p>
    </div>
  </div>,
  <div
    className="StyledBox-sc-13pk1d4-0 bZxUit"
  >
    <div
      className="StyledBox-sc-13pk1d4-0 hPokmc"
    >
      <img
        src="http://dummyimage.com/50x70.png/dddddd/000000"
      />
    </div>
    <div
      className="StyledBox-sc-13pk1d4-0 WHhky"
    >
      <small>
        Title:
      </small>
      <p>
        Mrs. Soffel
      </p>
    </div>
    <div
      className="StyledBox-sc-13pk1d4-0 bbVmir"
    >
      <small>
        Price:
      </small>
      <p>
        $5.48
      </p>
    </div>
    <div
      className="StyledBox-sc-13pk1d4-0 bbVmir"
    >
      <small>
        Quantity:
      </small>
      <p>
        <button
          className="StyledButton-sc-323bzc-0 fYqBJk"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          type="button"
        >
          -
        </button>
        2
        <button
          className="StyledButton-sc-323bzc-0 fYqBJk"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          type="button"
        >
          +
        </button>
      </p>
    </div>
    <div
      className="StyledBox-sc-13pk1d4-0 bbVmir"
    >
      <small>
        Action:
      </small>
      <p>
        <button
          className="StyledButton-sc-323bzc-0 fErzFK"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          type="button"
        >
          Remove
        </button>
      </p>
    </div>
  </div>,
  <div
    className="StyledBox-sc-13pk1d4-0 bZxUit"
  >
    <div
      className="StyledBox-sc-13pk1d4-0 hPokmc"
    >
      <img
        src="http://dummyimage.com/50x70.png/ff4444/ffffff"
      />
    </div>
    <div
      className="StyledBox-sc-13pk1d4-0 WHhky"
    >
      <small>
        Title:
      </small>
      <p>
        The Hound of the Baskervilles
      </p>
    </div>
    <div
      className="StyledBox-sc-13pk1d4-0 bbVmir"
    >
      <small>
        Price:
      </small>
      <p>
        $8.87
      </p>
    </div>
    <div
      className="StyledBox-sc-13pk1d4-0 bbVmir"
    >
      <small>
        Quantity:
      </small>
      <p>
        <button
          className="StyledButton-sc-323bzc-0 fYqBJk"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          type="button"
        >
          -
        </button>
        1
        <button
          className="StyledButton-sc-323bzc-0 fYqBJk"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          type="button"
        >
          +
        </button>
      </p>
    </div>
    <div
      className="StyledBox-sc-13pk1d4-0 bbVmir"
    >
      <small>
        Action:
      </small>
      <p>
        <button
          className="StyledButton-sc-323bzc-0 fErzFK"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          type="button"
        >
          Remove
        </button>
      </p>
    </div>
  </div>,
]
`;
